### ISSUE TITLE
Manually raising window without interacting with content

### ISSUE THREAD

**Issue by BatmanAoD** [joelpurra/xmouse-controls#26]
I'm guessing this is not possible without actually capturing mouse/keyboard events (and therefore not in scope for this project), but I thought I'd ask anyway: is it possible to manually raise a window without needing to click on it? I've found that most Windows apps have very few places to click that don't change _something_ (e.g. pausing a video, moving the mouse cursor & selection, etc) in the state of the app itself. It would be really nice if I could raise the currently-focused window using a keyboard shortcut or something without any risk of changing the app state.

**Comment by binki** [comment]
Almost any keystroke would risk changing the app state as apps can respond to mostly-benign keys like Ctrl if they really want to.

I have the same experience. If the app does not capture keyboard, you can alt+{tab, shift+tab} to raise it. This is closest to being reliable and a keyboard shortcut and is something I occasionally do.

In terms of finding benign places to click in apps, apps with title bars are best. The title bar will raise most windows even if clicking inside of the window doesn’t raise that window (yes, there are applications which are already hard to raise).

Another almost benign click target is scrollbars. I find that clicking a scrollbar will leave text fields focused, etc.

**Comment by gazialankus** [comment]
Yes the window can respond, but can we still have a keystorke that raises the window that I'm hovering on? I'm happy with having to find an unused keyboard shortcut, I just want this to be possible. I would be even happier to assign a mouse key to this, the forward and backward keys on my mouse are rarely used really. 

So, would you consider adding this feature to X-Mouse controls? 

[X] Trigger key (activate upon keypress rather than delay) [select key]

**Comment by joelpurra** [comment]
@BatmanAoD: if moving the mouse cursor is considered changing the app state, then changing the window z-order is definitely changing the app state, right?

Am not on Windows today, nor did I explore the builtin keyboard shortcuts in Windows 11 so far, but there's (finally?) official keyboard shortcut documentation online.

- https://support.microsoft.com/en-us/windows/keyboard-shortcuts-in-windows-dcc61a57-8ff0-cffe-9796-cb9706c75eec
- https://support.microsoft.com/en-us/windows/keyboard-shortcuts-in-apps-139014e7-177b-d1f3-eb2e-7298b2599a34
- https://support.microsoft.com/en-us/windows/windows-keyboard-shortcuts-for-accessibility-021bcb62-45c8-e4ef-1e4f-41b8c1fc87fd

Perhaps there's already something?

- <kbd>Alt-Esc</kbd> "Cycle through items in the order in which they were opened." A classic, far from as well-known as <kbd>Alt-Tab</kbd>.
- <kbd>F6</kbd> "Cycle through screen elements in a window or on the desktop." May trigger raising the window if input focus moves.
- <kbd>F4</kbd> "Display the items in the active list." Could perhaps randomly shake life to the hovered window if it uses standard user interface components.

How about simply right-clicking, does that work? Since it'd usually trigger displaying a right-click menu, the click itself presumably doesn't do anything else. Even if there's no menu for the location of the right-click, I'd expect it to be treated as a no-op anyway. There are probably empty/unresponsive/benign areas, to be separately discovered I guess, for right-click just as there are for left-click. Same goes for mouse button 3, 4, 5, 6 and higher, which are less used as @gazialankus mentions.

**Comment by joelpurra** [comment]
> So, would you consider adding this feature to X-Mouse controls?
> 
> [X] Trigger key (activate upon keypress rather than delay) [select key]

@gazialankus: sorry, this is out of scope for X-Mouse Controls. The program is not meant to run all the time, only when the settings need to be changed -- perhaps only once per Windows installation ;)

Also, I'm worried that attempting to control window order would be riddled with "unexpected behavior", in addition to what is seen already when using xmouse (in particular with a very low delay).

**Comment by BatmanAoD** [comment]
@joelpurra Sorry, I meant "text cursor", not "mouse cursor". That is, I don't like clicking to raise a window, only to see that I've moved a text cursor a dramatic distance from wherever I left it.

Sadly none of those suggested shortcuts do quite what I'd like (`F6` sounded promising but doesn't actually raise the window at all!). Right-click is indeed about the closest.

Thanks anyway!
